DataFormats/Detectors/TRD/include/DataFormatsTRD/T0FitHistos.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file T0FitHistos.h
+/// \brief Class to store the TRD PH values for each TRD chamber
+
+#ifndef ALICEO2_T0FITHISTOS_H
+#define ALICEO2_T0FITHISTOS_H
+
+#include "DataFormatsTRD/Constants.h"
+#include "DataFormatsTRD/PHData.h"
+#include "Framework/InputRecord.h"
+#include "Rtypes.h"
+#include <vector>
+#include <memory>
+#include <gsl/span>
+
+namespace o2
+{
+namespace trd
+{
+
+class T0FitHistos
+{
+ public:
+  T0FitHistos() = default;
+  T0FitHistos(const T0FitHistos&) = default;
+  ~T0FitHistos() = default;
+  void init();
+  void reset();
```

Comment:
for `reset()` I did not check, but is it used anywhere?

Reply:
I added it because it was implemented in both the vdrift and gain calibration container classes. But as far as I can tell, it is not used by the base class

Reply:
Correction: There is a reset() method in DetectorCalibration/TimeSlotCalibration.h:

```
virtual void reset()
  { // reset to virgin state (need for start - stop - start)
    mSlots.clear();
    mLastClosedTF = 0;
    mFirstTF = 0;
    mMaxSeenTF = 0;
    mLastCheckedTFInfiniteSlot = 0;
    mWasCheckedInfiniteSlot = false;
    initOutput();
  }
```

So it is used, and at the moment redefined by the derived class.
Should I leave it in or remove and let it use the base method?

Reply:
These methods are not related. The `TimeSlotCalibration` is the 'manager class' which handles all given slots and performs the calibration when some slots have enough statistics reached for example. The `reset` handles the case when a run is stopped and a new one started. In that case all the remaining slots are removed. Your `T0FitHistos` class is the 'container class'. Each slot knows its first and last seen TF, some statistics and one container. The container is never reset as far as I can see. You can merge it with other containers or remove it or create new ones. You can check also `Detectors/Calibration/include/DetectorsCalibration/TimeSlot.h` where the required functionality for a container class is defined

Reply:
ok, so it's not used and as per the readme only print, merge and fill need to be implemented. I removed reset()